s3transfer/manager.py

```diff
@@ -27,26 +29,40 @@
 class TransferConfig(object):
     def __init__(self,
                  multipart_threshold=8 * MB,
-                 max_concurrency=10,
                  multipart_chunksize=8 * MB,
-                 max_queue_size=0,
+                 max_request_concurrency=10,
```

Comment:
I was planning to have a compatibility layer anyways that will do some conversion such that if they gave me the deprecated Config object, then I would convert it to the appropriate variable. I have already made a need to add this compatibility layer anyways with the way subscribers are provided (i.e. they have to be of a certain class abstraction and you can provide more than one subscriber).

I feel like this name is much more clear on what this actually tunes given that I have three executors/pools of threads and was worth the change. What do you think?